packages/components/docs/jest.md

```diff
@@ -0,0 +1,113 @@
+# Jest Testing
+
+[Jest](https://jestjs.io/docs/en/getting-started.html) is a JavaScript Testing Framework that we use for unit testing
+our React components, models, and utility functions. It is in use in this labkey-ui-components package along with
+several LabKey modules that have React pages.
+
+Jest tests are quick to run and are a good tool for doing unit testing of React components, properties and methods of
+a model, utility functions, among other things. Jest tests in conjunction with other integration testing options (like
+the [@labkey/test](../../test/README.md) package and Selenium
+testing) can help us to get the full test coverage that we need for LabKey features / stories.
+
+With how quickly jest tests can run in comparison to Selenium tests and how much easier they are to maintain, it is
+good to make an automated test plan for your feature / story which will get us the coverage we want in the quickest
+turn around time possible. One good example of where a Selenium test is preffered over a jest test is when one component
+can affect another component in the application (i.e. based on state/data stored in the database that is changed by
+one component and shown in another).
+
```

Comment:
I'm not sure how helpful to others this is, but one of the things I struggled with in first ramping up on Jest tests was the shift away from an integration testing mindset—as in, I would be trying to test if I could click this and then detect this changed text/UI on the page, rather than seeing the test as essentially an exercise of props and state, and how changes those affects the component. 

I'd be fine with running the suggestion below by our other frontend gurus in case they have edits or comments on this test-writing approach. I'm guesstimating the intended audience based on the information level of this doc, but I think it might be appropriate to include a paragraph here that's something like:

In order to understand what scenarios to cover in writing unit tests for a React component, consider the set of meaningfully different combinations of props and state that will affect the composition of the component, as well as what actions within the component will affect its state. 
Ensuring that the appearance and behavior of the component is as expected for distinct batches of props, and that manipulation of internal state executes and behaves as desired, covers almost all that is recommendable in component unit testing. 

OPTIONAL?:
For example, rather than attempting to write a unit test that checks whether filling out a form and clicking a button displays a certain modal, divide the structure of this test into its component-based subsections. Check that manipulating the state of the controlled component handling the form populates the form fields as desired. Check that clicking the button properly mutates the state handling whether or not the modal is displayed. Separately, moving onto unit tests for the modal component, check that the correct modal is rendered depending on what props it is given.

Reply:
good suggestion, added.